hazelcast/src/main/java/com/hazelcast/topic/impl/TopicService.java

```diff
@@ -146,10 +147,14 @@ public void incrementReceivedMessages(String topicName) {
         getLocalTopicStats(topicName).incrementReceives();
     }
 
-    public void publishEvent(String name, TopicEvent event, boolean multithreaded) {
-        Collection<EventRegistration> registrations = eventService.getRegistrations(TopicService.SERVICE_NAME, name);
-        int partitionId = multithreaded ? counter.incrementAndGet() : name.hashCode();
-        eventService.publishEvent(TopicService.SERVICE_NAME, registrations, event, partitionId);
+    public void publishMessage(String topicName, Object payload, boolean multithreaded) {
+        Collection<EventRegistration> registrations = eventService.getRegistrations(SERVICE_NAME, topicName);
+        if (!registrations.isEmpty()) {
```

Comment:
This is slightly racy, isn't it? I know that there has to be a cut-off point, but still, there' some logic in between, that probably should be minimised?